In [1]:
import pandas as pd
import numpy as np

In [2]:
# Load excel file
xl = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', None)

In [3]:
xl.keys()

dict_keys(['Read_Me', ' Variable List', 'Supplemental Data - County', 'Supplemental Data - State', 'ACCESS', 'STORES', 'RESTAURANTS', 'ASSISTANCE', 'INSECURITY', 'TAXES', 'LOCAL', 'HEALTH', 'SOCIOECONOMIC'])

In [4]:
# Load separate excel sheets as dataframes
var_list = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', ' Variable List')
supp_count = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'Supplemental Data - County')
supp_state = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'Supplemental Data - State')

In [5]:
# Load separate excel sheets as dataframes
access = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'ACCESS')
stores = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'STORES')
rest = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'RESTAURANTS')
assist = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'ASSISTANCE')
insec = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'INSECURITY')
tax = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'TAXES')
local = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'LOCAL')
health = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'HEALTH')
soc_ec = pd.read_excel(r'./../data/FoodEnvironmentAtlas.xls', 'SOCIOECONOMIC')

In [6]:
# storing dataframe names and excel sheet names
dfs = [access, stores, rest, assist, insec, tax, local, health, soc_ec]
df_names = ['ACCESS', 'STORES', 'RESTAURANTS', 'ASSISTANCE', 'INSECURITY', 'TAXES', 'LOCAL', 'HEALTH', 'SOCIOECONOMIC']

In [7]:
# All dfs have same number of counties 
for c in dfs:
    print(c.shape)

(3143, 44)
(3143, 39)
(3143, 19)
(3143, 57)
(3143, 9)
(3143, 8)
(3144, 100)
(3143, 14)
(3143, 18)


Only local has 3144 rows. As we will see in the next steps, its data type is also not correct. This is due to the null value in FIPS county. Since only one row has missing county number and because other dataframes all have 3143 counties, I will remove the one row that has the missing value.

In [8]:
#  Check data type
# All FIPS columns should be int
for c in dfs:
    print(c.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3143 entries, 0 to 3142
Data columns (total 44 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   FIPS                       3143 non-null   int64  
 1   State                      3143 non-null   object 
 2   County                     3143 non-null   object 
 3   LACCESS_POP10              3143 non-null   float64
 4   LACCESS_POP15              3124 non-null   float64
 5   PCH_LACCESS_POP_10_15      3117 non-null   float64
 6   PCT_LACCESS_POP10          3143 non-null   float64
 7   PCT_LACCESS_POP15          3124 non-null   float64
 8   LACCESS_LOWI10             3143 non-null   float64
 9   LACCESS_LOWI15             3123 non-null   float64
 10  PCH_LACCESS_LOWI_10_15     3115 non-null   float64
 11  PCT_LACCESS_LOWI10         3143 non-null   float64
 12  PCT_LACCESS_LOWI15         3123 non-null   float64
 13  LACCESS_HHNV10             3143 non-null   float

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3143 entries, 0 to 3142
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   FIPS               3143 non-null   int64  
 1   State              3143 non-null   object 
 2   County             3143 non-null   object 
 3   SODATAX_STORES14   3143 non-null   float64
 4   SODATAX_VENDM14    3143 non-null   float64
 5   CHIPSTAX_STORES14  3143 non-null   float64
 6   CHIPSTAX_VENDM14   3143 non-null   float64
 7   FOOD_TAX14         3143 non-null   float64
dtypes: float64(5), int64(1), object(2)
memory usage: 196.6+ KB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3144 entries, 0 to 3143
Data columns (total 100 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   FIPS                         3143 non-null   float64
 1   State                        3143 non-null   object 
 2   

In [9]:
print(local['FIPS'].isnull().sum())
# Removing row with no FIPS code in the dataframe 'local'
local.dropna(axis=0, subset=['FIPS'], inplace=True)
local['FIPS'] = pd.to_numeric(local['FIPS']).astype('int64')  #change to int64

1


In [10]:
# Check result
# local.info()

In [11]:
# Investigating variables in the dataset
for c in df_names:
    print(var_list[var_list['Category Code'] == str(c)])

                            Category Name Category Code     Subcategory Name  \
0   Access and Proximity to Grocery Store        ACCESS              Overall   
1   Access and Proximity to Grocery Store        ACCESS              Overall   
2   Access and Proximity to Grocery Store        ACCESS              Overall   
3   Access and Proximity to Grocery Store        ACCESS              Overall   
4   Access and Proximity to Grocery Store        ACCESS              Overall   
5   Access and Proximity to Grocery Store        ACCESS  Household Resources   
6   Access and Proximity to Grocery Store        ACCESS  Household Resources   
7   Access and Proximity to Grocery Store        ACCESS  Household Resources   
8   Access and Proximity to Grocery Store        ACCESS  Household Resources   
9   Access and Proximity to Grocery Store        ACCESS  Household Resources   
10  Access and Proximity to Grocery Store        ACCESS  Household Resources   
11  Access and Proximity to Grocery Stor

265    CNTY10         % change  
                     Category Name  Category Code Subcategory Name  \
266  Socioeconomic Characteristics  SOCIOECONOMIC       Race & Age   
267  Socioeconomic Characteristics  SOCIOECONOMIC       Race & Age   
268  Socioeconomic Characteristics  SOCIOECONOMIC       Race & Age   
269  Socioeconomic Characteristics  SOCIOECONOMIC       Race & Age   
270  Socioeconomic Characteristics  SOCIOECONOMIC       Race & Age   
271  Socioeconomic Characteristics  SOCIOECONOMIC       Race & Age   
272  Socioeconomic Characteristics  SOCIOECONOMIC       Race & Age   
273  Socioeconomic Characteristics  SOCIOECONOMIC       Race & Age   
274  Socioeconomic Characteristics  SOCIOECONOMIC     Income Level   
275  Socioeconomic Characteristics  SOCIOECONOMIC     Income Level   
276  Socioeconomic Characteristics  SOCIOECONOMIC     Income Level   
277  Socioeconomic Characteristics  SOCIOECONOMIC     Income Level   
278  Socioeconomic Characteristics  SOCIOECONOMIC     Inc

In [12]:
# Show all columns
pd.set_option('max_columns', None)

In [13]:
# local.isnull().sum()

In [14]:
# Preparing dataframes for merge (only dataframes that are of interest)
# Remove common columns (State and County names) that would be double counted in the merge
df_remstate = [stores, rest, assist, insec, local, soc_ec]
for c in df_remstate:
    c.drop(labels=['State','County'], axis=1, inplace=True)

In [15]:
# Number of columns is reduced by 2 for all dfs except access
for c in df_remstate: 
    print(c.shape)

(3143, 37)
(3143, 17)
(3143, 55)
(3143, 7)
(3143, 98)
(3143, 16)


In [16]:
# Keep only needed variables

# Removing 23
access.drop(labels=['LACCESS_POP10', 'LACCESS_POP15', 'PCH_LACCESS_POP_10_15', 'LACCESS_LOWI10', 'LACCESS_LOWI15',
                    'PCH_LACCESS_LOWI_10_15', 'LACCESS_HHNV10', 'LACCESS_HHNV15', 'PCH_LACCESS_HHNV_10_15',
                    'LACCESS_SNAP15', 'LACCESS_CHILD10', 'LACCESS_CHILD15', 'LACCESS_CHILD_10_15', 
                    'LACCESS_SENIORS10', 'LACCESS_SENIORS15', 'PCH_LACCESS_SENIORS_10_15', 'LACCESS_WHITE15', 
                    'LACCESS_BLACK15', 'LACCESS_HISP15', 'LACCESS_NHASIAN15', 'LACCESS_NHNA15', 
                    'LACCESS_NHPI15', 'LACCESS_MULTIR15'], axis=1, inplace=True)
# Removing 26
stores.drop(labels=['GROC11', 'GROC16', 'PCH_GROC_11_16', 'PCH_GROCPTH_11_16', 'SUPERC11', 'SUPERC16',
                    'PCH_SUPERC_11_16', 'PCH_SUPERCPTH_11_16', 'CONVS11', 'CONVS16', 'PCH_CONVS_11_16', 
                    'PCH_CONVSPTH_11_16', 'SPECS11', 'SPECS16', 'PCH_SPECS_11_16', 'PCH_SPECSPTH_11_16',
                    'SNAPS12', 'SNAPS17', 'PCH_SNAPS_12_17', 'PCH_SNAPSPTH_12_17', 'WICS11', 'WICS16',
                    'PCH_WICS_11_16', 'WICSPTH11', 'WICSPTH16', 'PCH_WICSPTH_11_16'], axis=1, inplace=True)
# Removing 8
rest.drop(labels=['FFR11', 'FFR16', 'PCH_FFR_11_16', 'PCH_FFRPTH_11_16', 'FSR11', 'FSR16',
                  'PCH_FSR_11_16', 'PCH_FSRPTH_11_16'], axis=1, inplace=True)
# Removing 37 
assist.drop(labels=['REDEMP_SNAPS12', 'REDEMP_SNAPS17', 'PCH_REDEMP_SNAPS_12_17', 'PCH_SNAP_12_17',
                    'PC_SNAPBEN12', 'PC_SNAPBEN17', 'PCH_PC_SNAPBEN_12_17', 'SNAP_OAPP09',
                    'SNAP_OAPP16', 'SNAP_CAP09', 'SNAP_CAP16', 'SNAP_BBCE09', 'SNAP_BBCE16',
                    'SNAP_REPORTSIMPLE09', 'SNAP_REPORTSIMPLE16', 'PCH_NSLP_12_17', 'PCH_SBP_12_17',
                    'PCH_SFSP_12_17', 'PC_WIC_REDEMP11', 'PC_WIC_REDEMP16', 'PCH_PC_WIC_REDEMP_11_16', 
                    'REDEMP_WICS11', 'REDEMP_WICS16', 'PCH_REDEMP_WICS_11_16', 'PCT_WIC12', 'PCT_WIC17',
                    'PCH_WIC_12_17', 'PCT_WICINFANTCHILD14', 'PCT_WICINFANTCHILD16', 'PCH_WICINFANTCHILD_14_16',
                    'PCT_WICWOMEN14', 'PCT_WICWOMEN16', 'PCH_WICWOMEN_14_16', 'PCT_CACFP12', 'PCT_CACFP17',
                    'PCH_CACFP_12_17', 'PCH_FDPIR_12_15'], axis=1, inplace=True)
# Removing 4
insec.drop(labels=['CH_FOODINSEC_14_17', 'VLFOODSEC_12_14', 'VLFOODSEC_15_17',
                   'CH_VLFOODSEC_14_17'], axis=1, inplace=True)
# Removing 92
local.drop(labels=['PCH_DIRSALES_FARMS_07_12', 'PCT_LOCLFARM07', 'PCT_LOCLFARM12', 'PCT_LOCLSALE07',
                   'PCT_LOCLSALE12', 'DIRSALES07', 'DIRSALES12', 'PCH_DIRSALES_07_12', 'PC_DIRSALES07',
                   'PC_DIRSALES12', 'PCH_PC_DIRSALES_07_12','FMRKT13', 'FMRKT18', 'PCH_FMRKT_13_18',
                   'PCH_FMRKTPTH_13_18', 'FMRKT_SNAP18', 'PCT_FMRKT_SNAP18', 'FMRKT_WIC18',
                   'PCT_FMRKT_WIC18', 'FMRKT_WICCASH18', 'PCT_FMRKT_WICCASH18', 'FMRKT_SFMNP18',
                   'PCT_FMRKT_SFMNP18','FMRKT_CREDIT18', 'PCT_FMRKT_CREDIT18', 'FMRKT_FRVEG18',
                   'PCT_FMRKT_FRVEG18', 'FMRKT_ANMLPROD18', 'PCT_FMRKT_ANMLPROD18', 'FMRKT_BAKED18',
                   'PCT_FMRKT_BAKED18', 'FMRKT_OTHERFOOD18', 'PCT_FMRKT_OTHERFOOD18', 'VEG_FARMS07',
                   'VEG_FARMS12','PCH_VEG_FARMS_07_12', 'VEG_ACRES07', 'VEG_ACRES12','PCH_VEG_ACRES_07_12',
                   'VEG_ACRESPTH07', 'VEG_ACRESPTH12', 'PCH_VEG_ACRESPTH_07_12', 'FRESHVEG_FARMS07',
                   'FRESHVEG_FARMS12', 'PCH_FRESHVEG_FARMS_07_12', 'FRESHVEG_ACRES07', 'FRESHVEG_ACRES12',
                   'PCH_FRESHVEG_ACRES_07_12', 'FRESHVEG_ACRESPTH07', 'FRESHVEG_ACRESPTH12',
                   'PCH_FRESHVEG_ACRESPTH_07_12', 'ORCHARD_FARMS07', 'ORCHARD_FARMS12', 
                   'PCH_ORCHARD_FARMS_07_12', 'ORCHARD_ACRES07', 'ORCHARD_ACRES12', 'PCH_ORCHARD_ACRES_07_12',
                   'ORCHARD_ACRESPTH07', 'ORCHARD_ACRESPTH12', 'PCH_ORCHARD_ACRESPTH_07_12', 'BERRY_FARMS07',
                   'BERRY_FARMS12', 'PCH_BERRY_FARMS_07_12', 'BERRY_ACRES07', 'BERRY_ACRES12', 
                   'PCH_BERRY_ACRES_07_12', 'BERRY_ACRESPTH07', 'BERRY_ACRESPTH12', 
                   'PCH_BERRY_ACRESPTH_07_12', 'SLHOUSE07', 'SLHOUSE12', 'PCH_SLHOUSE_07_12',
                   'GHVEG_FARMS07', 'GHVEG_FARMS12', 'PCH_GHVEG_FARMS_07_12', 'GHVEG_SQFT07', 'GHVEG_SQFT12',
                   'PCH_GHVEG_SQFT_07_12', 'GHVEG_SQFTPTH07', 'GHVEG_SQFTPTH12', 'PCH_GHVEG_SQFTPTH_07_12',
                   'CSA07', 'CSA12', 'PCH_CSA_07_12', 'AGRITRSM_OPS07', 'AGRITRSM_OPS12', 
                   'PCH_AGRITRSM_OPS_07_12', 'AGRITRSM_RCT07', 'AGRITRSM_RCT12', 'PCH_AGRITRSM_RCT_07_12', 
                   'FARM_TO_SCHOOL13', 'FARM_TO_SCHOOL15'], axis=1, inplace=True)

# Removing 3
soc_ec.drop(labels=['PERPOV10','PERCHLDPOV10','POPLOSS10'], axis=1, inplace=True)

In [17]:
for c in df_remstate:
    print(c.shape)

(3143, 11)
(3143, 9)
(3143, 18)
(3143, 3)
(3143, 6)
(3143, 13)


In [18]:
# Checking uniqueness of county codes
access['FIPS'].nunique()

3143

In [19]:
# Merging 7 dataframes of interest on FIPS code unique to counties
df_comb = access.merge(stores,on='FIPS').merge(rest,on='FIPS').merge(assist,on='FIPS').merge(
    insec,on='FIPS').merge(local,on='FIPS').merge(soc_ec,on='FIPS')
df_comb.head()

FIPS State   County  PCT_LACCESS_POP10  PCT_LACCESS_POP15  \
0  1001    AL  Autauga          33.769657          32.062255   
1  1003    AL  Baldwin          19.318473          16.767489   
2  1005    AL  Barbour          20.840972          22.105560   
3  1007    AL     Bibb           4.559753           4.230324   
4  1009    AL   Blount           2.700840           6.497380   

   PCT_LACCESS_LOWI10  PCT_LACCESS_LOWI15  PCT_LACCESS_HHNV10  \
0            9.793530           11.991125            3.284786   
1            5.460261            5.424427            2.147827   
2           11.420316           10.739667            4.135869   
3            2.144661            2.601627            3.458580   
4            1.062468            2.880150            3.269380   

   PCT_LACCESS_HHNV15  PCT_LACCESS_SNAP15  PCT_LACCESS_CHILD10  \
0            3.351332            4.608749             8.837112   
1            1.905114            1.298900             4.343199   
2            4.329378            4.303147             3.425062   
3            2.821427            0.676710             1.087518   
4            3.336414            0.812727             0.671490   

   PCT_LACCESS_CHILD15  PCT_LACCESS_SENIORS10  PCT_LACCESS_SENIORS15  \
0             8.460485               4.376378               3.996279   
1             3.844936               3.513570               3.061840   
2             3.758341               2.805166               3.001695   
3             1.015242               0.657008               0.600865   
4             1.588720               0.340269               0.882583   

   PCT_LACCESS_WHITE15  PCT_LACCESS_BLACK15  PCT_LACCESS_HISP15  \
0            23.163613             7.726582            0.863345   
1            13.981393             1.942757            0.755973   
2            10.302934             9.881360            1.855183   
3             2.358770             1.828933            0.037516   
4             5.909147             0.167201            0.867886   

   PCT_LACCESS_NHASIAN15  PCT_LACCESS_NHNA15  PCT_LACCESS_NHPI15  \
0               0.159000            0.112092            0.016159   
1               0.116833            0.099662            0.008131   
2               0.062266            0.145539            0.029436   
3               0.008703            0.010967            0.000000   
4               0.014705            0.050484            0.001854   

   PCT_LACCESS_MULTIR15  GROCPTH11  GROCPTH16  SUPERCPTH11  SUPERCPTH16  \
0              0.884808   0.090581   0.054271     0.018116     0.018090   
1              0.618712   0.144746   0.139753     0.032166     0.033733   
2              1.684025   0.219370   0.155195     0.000000     0.038799   
3              0.022951   0.263794   0.220916     0.043966     0.044183   
4              0.353990   0.121608   0.086863     0.017373     0.017373   

   CONVSPTH11  CONVSPTH16  SPECSPTH11  SPECSPTH16  SNAPSPTH12  SNAPSPTH17  \
0    0.561604    0.560802    0.018116    0.018090    0.674004    0.804747   
1    0.573622    0.568650    0.107219    0.130115    0.725055    0.890836   
2    0.804358    0.737177    0.109685    0.077598    1.280590    1.424614   
3    0.835348    0.662749    0.000000    0.000000    0.719122    0.801423   
4    0.521177    0.469059    0.017373    0.000000    0.657144    0.692374   

   FFRPTH11  FFRPTH16  FSRPTH11  FSRPTH16  PC_FFRSALES07  PC_FFRSALES12  \
0  0.615953  0.795977  0.579721  0.560802     649.511367      674.80272   
1  0.648675  0.751775  1.157966  1.137300     649.511367      674.80272   
2  0.694673  0.892372  0.621549  0.543183     649.511367      674.80272   
3  0.263794  0.309283  0.219829  0.309283     649.511367      674.80272   
4  0.347451  0.399569  0.260589  0.208471     649.511367      674.80272   

   PC_FSRSALES07  PC_FSRSALES12  PCT_SNAP12  PCT_SNAP17  SNAP_PART_RATE11  \
0     484.381507     512.280987   18.908476   16.500056             84.02   
1     484.381507     512.280987   18.908476   16.500056           

In [20]:
df_comb.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3143 entries, 0 to 3142
Data columns (total 75 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   FIPS                   3143 non-null   int64  
 1   State                  3143 non-null   object 
 2   County                 3143 non-null   object 
 3   PCT_LACCESS_POP10      3143 non-null   float64
 4   PCT_LACCESS_POP15      3124 non-null   float64
 5   PCT_LACCESS_LOWI10     3143 non-null   float64
 6   PCT_LACCESS_LOWI15     3123 non-null   float64
 7   PCT_LACCESS_HHNV10     3143 non-null   float64
 8   PCT_LACCESS_HHNV15     3140 non-null   float64
 9   PCT_LACCESS_SNAP15     3123 non-null   float64
 10  PCT_LACCESS_CHILD10    3143 non-null   float64
 11  PCT_LACCESS_CHILD15    3124 non-null   float64
 12  PCT_LACCESS_SENIORS10  3143 non-null   float64
 13  PCT_LACCESS_SENIORS15  3124 non-null   float64
 14  PCT_LACCESS_WHITE15    3124 non-null   float64
 15  PCT_

Total columns is (21+11+9+18+3+6+13)-6=75